### PR TITLE
RavenDB-4986: Check & Alert if swapping on hdd instead of ssd

### PR DESCRIPTION
**Pull request by aviviadi**

### CODE REVIEW

File: src/Raven.Server/ServerWide/ServerStore.cs [hunk 1]
```diff
@@ -247,6 +247,33 @@ public void Initialize()
...
+            try
+            {
+                if (MemoryInformation.IsSwappingOnHddInsteadOfSsd())
+                {
+                    var alert = AlertRaised.Create("Swap Storage Type Warning",
+                        "OS swapping on at least one HDD drive while there is at least one SSD drive on this system. " +
+                        "This can cause a slowdown, consider moving swap-partition/pagefile to SSD",
+                        AlertType.SwappingHddInsteadOfSsd,
+                        NotificationSeverity.Warning,
+                        "Swap Storage Type Warning");
```
Comment (ml054) [inline src/Raven.Server/ServerWide/ServerStore.cs:259]: Please delete this argument - it is not needed here.

Comment (aviviadi) [reply src/Raven.Server/ServerWide/ServerStore.cs:259]: The argument is copied from v3.5, what do you suggest instead?

Comment (ml054) [reply src/Raven.Server/ServerWide/ServerStore.cs:259]: Simply omit latest parameter, since it is optional parameter (with default = null)

Comment (aviviadi) [reply src/Raven.Server/ServerWide/ServerStore.cs:259]: ohh I see. You mean the key string. But isn't it important to specify a unique key there ? (in order not to get multiple notifications in Studio)

Comment (ml054) [reply src/Raven.Server/ServerWide/ServerStore.cs:259]: AlertType.SwappingHddInsteadOfSsd already gives you uniqueness. The last parameter is used when you want to distinguish operations which can be multiple, like: in SQL replication we set type to AlertType.SqlConnection error and then as last parameter we pass sql replication setup name. As result unique key is created using 2 fields.

Comment (aviviadi) [reply src/Raven.Server/ServerWide/ServerStore.cs:259]: https://github.com/ravendb/ravendb/pull/2512

**Pull request merged**